Update GetHashCode implementations to use HashCode.Combine (#52121)
dotnet/docs: docs/fundamentals/runtime-libraries/system-object-gethashcode.md

:::code language="fsharp" source="./snippets/System/Object/GetHashCode/fsharp/xor1.fs" id="Snippet2":::
:::code language="vb" source="./snippets/System/Object/GetHashCode/vb/xor1.vb" id="Snippet2":::

The previous example returns the same hash code for (n1, n2) and (n2, n1), and so may generate more collisions than are desirable. The recommended solution is to use <xref:System.HashCode.Combine%2A?displayProperty=nameWithType>, which is available starting with .NET Core 2.1. It avoids the symmetry problem and produces a well-distributed hash code without the overhead of creating a `Tuple` object.

:::code language="csharp" source="./snippets/System/Object/GetHashCode/csharp/xor2.cs" id="Snippet3":::
:::code language="fsharp" source="./snippets/System/Object/GetHashCode/fsharp/xor2.fs" id="Snippet3":::
:::code language="vb" source="./snippets/System/Object/GetHashCode/vb/xor2.vb" id="Snippet3":::

> Before `HashCode.Combine` was available, an alternative was to weight the individual hash codes by left-shifting the hash codes of successive fields by two or more bits. Optimally, bits shifted beyond bit 31 should wrap around rather than be discarded. Since bits are discarded by the left-shift operators in both C# and Visual Basic, this requires creating a left shift-and-wrap method like the following:

Reviewer: Change to say this is framework only:

```suggestion
In .NET Framework, an alternative is to weight the individual hash codes by left-shifting the hash codes of successive fields by two or more bits. Optimally, bits shifted beyond bit 31 should wrap around rather than be discarded. Since bits are discarded by the left-shift operators in both C# and Visual Basic, this requires creating a left shift-and-wrap method like the following:
```
